### ISSUE TITLE
No output for canvas today

### ISSUE THREAD

**Issue by pensionado** [tvgrabbers/tvgrabpyAPI#28]
After the daily run, again no xml output for canvas (like reported earlier 16/11). Only thing I can see in the logs (line 4308) is that just after it says that it has 84 programs, it removes one with same stop and start time (line 4326) and I suspect this is the first of the total list of programs, seen the time of the program, start of day this morning (could that be the bug ?). Log and xml output attached.
[Documents_1.zip](https://github.com/tvgrabbers/tvgrabpyAPI/files/1537987/Documents_1.zip)

**Comment by hikavdh** [comment]
I think I know what happens. Every programme is an object that knows which programme comes before and after. Also the channel object knows which programme is the first and the last one. When I parse the list to create the output I jump to the first and then to the next until there is no next. When I remove a programme because it is empty I link the previous to the next, only if the removed programme is the first there is no previous and I forgot to check on that event, So in that case the pointer to the first programme for that channel will be empty. I will fix this in the next release.

**Comment by hikavdh** [comment]
If you want to fix it in your own set:
tv_grab_channel.py at line 1592 add (the lines starting with a +):
```
            if isinstance(ngap, GapNode):
                self.remove_gap(ngap)

+            if pn == self.first_node:
+                self.first_node = nnode
+
+            if pn == self.last_node:
+                self.last_node = pnode
+
            gap = self.link_nodes(pnode, nnode)
            if isinstance(pnode, ProgramNode):
                pnode.next_gap = gap

```

**Comment by hikavdh** [comment]
Added fix to 1.0.8 beta
